protocol/messenger_activity_center.go

```diff
@@ -60,16 +63,70 @@ func (m *Messenger) ActivityCenterNotificationsCount(request ActivityCenterCount
 }
 
 func (m *Messenger) HasUnseenActivityCenterNotifications() (bool, error) {
-	return m.persistence.HasUnseenActivityCenterNotifications()
+	seen, _, err := m.persistence.HasUnseenActivityCenterNotifications()
+	return seen, err
 }
 
 func (m *Messenger) GetActivityCenterState() (*ActivityCenterState, error) {
 	return m.persistence.GetActivityCenterState()
 }
 
+func (m *Messenger) syncActivityCenterNotificationState(state *ActivityCenterState) error {
+	if state == nil {
+		return nil
+	}
+	syncStateMessage := &protobuf.SyncActivityCenterNotificationState{
+		UpdatedAt: state.UpdatedAt,
+		HasSeen:   state.HasSeen,
+	}
+	encodedMessage, err := proto.Marshal(syncStateMessage)
+	if err != nil {
+		return err
+	}
+	return m.sendToPairedDevices(context.TODO(), common.RawMessage{
+		Payload:             encodedMessage,
+		MessageType:         protobuf.ApplicationMetadataMessage_SYNC_ACTIVITY_CENTER_NOTIFICATION_STATE,
+		ResendAutomatically: true,
+	})
+}
+
+func (m *Messenger) syncActivityCenterNotifications(notifications []*ActivityCenterNotification) (err error) {
+	if notifications == nil {
```

Comment:
@qfrank this check here is unnecessary, you can remove it.

Reply:
I think it make sense in some cases, but i can't remember which case ATM .. check is always good than not 🙂